shared/util/local_processor.go

```diff
@@ -34,11 +35,19 @@ func (s *LocalProcessor) Tarzip(src string, f *os.File) error {
 	defer tw.Close()
 
 	return filepath.Walk(src, func(file string, fi os.FileInfo, err error) error {
```

Comment:
How will this work with directories? If there is a .terraform directory does it not go into that directory? I think if you want to skip a directory, you will need to do something like:
~~~golang
if info.IsDir() && info.Name() == subDirToSkip {
  fmt.Printf("skipping a dir without errors: %+v \n", info.Name())
  return filepath.SkipDir
}
~~~

Reply:
Walk() is not recursive. The way it works is that it loops through the entire fileset across the entire folder tree, and skipping a dir entry will not skip the rest of the files in it. Instead I look at the entire file path and get the top level directory from it. If it is not allowed, I skip the entry. This way all files under `.terraform` will be skipped.

Reply:
looks like you fixed it up, but I think it is recursive: https://pkg.go.dev/path/filepath#WalkFunc

Reply:
the error type controls the flow of how the walker walks

Reply:
Another revision.